ostd/libs/linux-bzimage/setup/src/x86/amd64_efi/decoder.rs

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module is used to decompress payload.
+
+extern crate alloc;
+
+use core2::io::Read;
+use libflate::gzip::Decoder;
+pub use alloc::vec::Vec;
+
+/// This function decompresses payload.
+pub fn decompress_payload(payload: &[u8]) -> Vec<u8> {
+    let mut decoder = Decoder::new(payload).unwrap();
+    let mut kernel = Vec::new();
+    decoder.read_to_end(&mut kernel).unwrap();
```

Comment:
I am afraid that allocating a large buffer (up to hundreds of MB) on the heap during the bootup stage is not a good idea. It is definitely not efficient (heap is best suited for small objects) and may be infeasible (the size of the initial statically allocated heap is quite limited).

Reply:
I checked the heap allocator enabled during the bootup stage, it seemed the heap would be enlarged when it ran out.